### PR TITLE
Stop scheduling local notification when there is an evergreen campaign

### PR DESCRIPTION
**Pull request by irfano**
Part of #12188 

### Description

This PR stops scheduling `blaze_no_campaign_reminder` type notification, if there is an endless campaign for the site. 
Internal discussion: p1725265460299019-slack-C03L1NF1EA3

### Steps to reproduce

To update 30 days to 5 seconds, apply this patch → [Trigger_blaze_reminder_notification_for_testing.patch](https://github.com/user-attachments/files/16874445/Trigger_blaze_reminder_notification_for_testing.patch)

After applying patch, if you open My Store tab with a site that has Blaze campaigns, it will schedule reminder notification.

### Testing information

- Test with a site that does not have any evergreen campaigns, It should display a notification.
- Test with a site that has an evergreen campaign, It shouldn't display a notification.

### The tests that have been performed

Explained above.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt [hunk 1]
```diff
@@ -43,6 +43,11 @@ class BlazeCampaignsObserver @Inject constructor(
         if (campaigns.isEmpty()) {
             // There are no campaigns. Skip scheduling the notification.
             return
+        } else if (campaigns.any { it.isEndlessCampaign }) {
+            // There is an active endless campaign.
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt]: Hmm, I might be missing something, but this doesn't check that the campaign is active, it only checks if there's an endless  campaign on the list of campaigns from the DB. So, if I have a `completed` endless campaign on the list the notification won't ever be scheduled?

Comment (irfano) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt]: The comment was wrong and I fixed it: ee0af1bb5d6d822ebb24eb710150c043387719d7
> So, if I have a completed endless campaign on the list the notification won't ever be scheduled?

Yes, we can't technically know when an endless campaign is cancelled or completed. So, we won't schedule any "no_campaign" reminders if there is an endless campaign.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt]: Hey @irfano I don't want to block this PR so I'll pre approve. 
I left a comment in Slack to keep the discussion going, as disabling the reminder when an endless campaign exists (wether is active or not) seems to drastic IMHO. 
p1725989620766459/1725265460.299019-slack-C03L1NF1EA3

**Review by JorgeMucientes — APPROVED** [review]

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt (hunk 1 above)
Comment (irfano) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt]: I am closing this PR and will open a new one based on our discussion.

**Pull request closed without merging**